st2tests/integration/mistral/test_filters.py

```diff
@@ -0,0 +1,373 @@
+            parameters={
+                "input_str": REGEX_SEARCH_STR_2,
+                "replacement_str": "foo"
+            }
+        )
+        execution = self._wait_for_completion(execution)
+        self._assert_success(execution, num_tasks=1)
+        self.assertEqual(
+            execution.result['result_jinja'],
+            "foo is your address. My address is foo."
+        )
+        self.assertEqual(execution.result['result_yaql'], "foo is your address. My address is foo.")
+
+
+class RegexSearchFiltersTest(base.TestWorkflowExecution):
+
+    def test_regex_search(self):
+        execution = self._execute_workflow('examples.mistral-customfilters-regex_search',
+                                           parameters={"input_str": REGEX_SEARCH_STR})
+        execution = self._wait_for_completion(execution)
+        self._assert_success(execution, num_tasks=1)
+        self.assertTrue(execution.result['result_jinja'])
+        self.assertTrue(execution.result['result_yaql'])
+
+    def test_regex_nosearch(self):
+        execution = self._execute_workflow('examples.mistral-customfilters-regex_search',
+                                           parameters={"input_str": REGEX_SEARCH_STR_3})
+        execution = self._wait_for_completion(execution)
+        self._assert_success(execution, num_tasks=1)
+        self.assertFalse(execution.result['result_jinja'])
+        self.assertFalse(execution.result['result_yaql'])
+
+
+class RegexSubstringFiltersTest(base.TestWorkflowExecution):
+
+    def test_regex_substring(self):
+        execution = self._execute_workflow('examples.mistral-customfilters-regex_substring',
+                                           parameters={"input_str": REGEX_SEARCH_STR})
+        execution = self._wait_for_completion(execution)
+        self._assert_success(execution, num_tasks=1)
+        self.assertEqual(execution.result['result_jinja'], '567 Elsewhere Dr')
+        self.assertEqual(execution.result['result_yaql'], '567 Elsewhere Dr')
+        self.assertEqual(execution.result['result_jinja_index_1'], '123 Somewhere Ave')
+        self.assertEqual(execution.result['result_yaql_index_1'], '123 Somewhere Ave')
+
+
+class ToHumanTimeFromSecondsFiltersTest(base.TestWorkflowExecution):
+
+    def test_to_human_time_from_seconds(self):
+        execution = self._execute_workflow(
+            'examples.mistral-customfilters-to_human_time_from_seconds',
+            parameters={"seconds": 4587}
+        )
+        execution = self._wait_for_completion(execution)
+        self._assert_success(execution, num_tasks=1)
+        self.assertEqual(execution.result['result_jinja'], '1h16m27s')
+        self.assertEqual(execution.result['result_yaql'], '1h16m27s')
+
+
+class UseNoneFiltersTest(base.TestWorkflowExecution):
+
+    def test_use_none(self):
+        inputs = {'input_str': 'foo'}
+        execution = self._execute_workflow(
+            'examples.mistral-customfilters-use_none', parameters=inputs
+        )
```

Comment:
Oops, now I remember. The underlying implementation actually explicitly checks for `None` (I mistakenly thought it was a simple `if not value:` but it is is literally `if value is None`) https://github.com/StackStorm/st2mistral/pull/30/files#diff-d0cfa0e285114587d5c640c4937d3d77R26

So to keep from modifying the underlying function (which would technically be a breaking change - or at least a deviation from the sister function in st2), I need to be able to supply `null` values, and I can't do that if the parameter type is `string`. Thoughts either way (as in changing this workflow to fit the existing function vs changing the way the function works)